mcproto/types/nbt.py

```diff
@@ -0,0 +1,1399 @@
+        # Compare the float values with a small epsilon
+        if type(self) is not type(other):
+            return False
+        other.payload = cast(float, other.payload)
+        if self.name != other.name:
+            return False
+        return abs(self.payload - other.payload) < 1e-6
+
+    @property
+    @override
+    def value(self) -> float:
+        """Get the float value of the FloatNBT tag."""
+        return self.payload
+
+
+class DoubleNBT(FloatNBT):
+    """NBT tag representing a double-precision floating-point value, represented as a 64-bit IEEE 754-2008 binary64."""
+
+    __slots__ = ()
+
+    @override
+    def write_to(self, buf: Buffer, with_type: bool = True, with_name: bool = True) -> None:
+        """Write the DoubleNBT tag to the buffer.
+
+        :param buf: The buffer to write to.
+        :param with_type: Whether to include the type of the tag in the serialization.
+        :param with_name: Whether to include the name of the tag in the serialization.
+
+        :note: The double value is written as a 64-bit floating-point value in big-endian format.
+        """
+        self._write_header(buf, with_type=with_type, with_name=with_name)
+        buf.write_value(StructFormat.DOUBLE, self.payload)
+
+    @override
+    @classmethod
+    def read_from(cls, buf: Buffer, with_type: bool = True, with_name: bool = True) -> DoubleNBT:
+        """Read the DoubleNBT tag from the buffer.
+
+        :param buf: The buffer to read from.
+        :param with_type: Whether to read the type of the tag from the buffer. If this is False, the type of the class
+            will be used.
+        :param with_name: Whether to read the name of the tag to the buffer as a TAG_String.
+
+        :return: The DoubleNBT tag.
+        """
+        name, tag_type = cls._read_header(buf, read_type=with_type, with_name=with_name)
+        if _get_tag_type(cls) != tag_type:
+            raise TypeError(f"Expected a {_get_tag_type(cls).name} tag, but found a different tag ({tag_type.name}).")
+
+        if buf.remaining < 8:
+            raise IOError("Buffer does not contain enough data to read a double.")
+
+        return DoubleNBT(buf.read_value(StructFormat.DOUBLE), name=name)
+
+
+class ByteArrayNBT(NBTag):
+    """NBT tag representing an array of bytes. The length of the array is stored as a signed 32-bit integer."""
+
+    __slots__ = ()
+
+    payload: bytes
+
+    @override
+    def write_to(self, buf: Buffer, with_type: bool = True, with_name: bool = True) -> None:
+        """Write the ByteArrayNBT tag to the buffer.
+
+        :param buf: The buffer to write to.
+        :param with_type: Whether to include the type of the tag in the serialization.
+        :param with_name: Whether to include the name of the tag in the serialization.
+
+        :note: The length of the byte array is written as a signed 32-bit integer in big-endian format.
```

Comment:
```suggestion
        .. note:: The length of the byte array is written as a signed 32-bit integer in big-endian format.
```

Applies for all sphinx directives. See [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html)